src/processes.c

```diff
@@ -283,6 +266,31 @@ int     getthrds64( pid_t, void *, int, tid64_t *, int );
 int getargs (void *processBuffer, int bufferLen, char *argsBuffer, int argsLen);
 #endif /* HAVE_PROCINFO_H */
 
+static void ps_copy_procstat_data (procstat_data_t *dst, procstat_data_t *src)
```

Comment:
This makes me wonder if we should've split this into two structs, one for the gauge metrics and one for the counter metrics... Then this function would be unnecessary, as you'd literally just say `*dst = *src` and copy by value.

Reply:
BTW, can we avoid using abbreviations like dst and src for readability? In this method, it's pretty obvious what they mean. But it somehow reminds me of the `ps` and `pse` stuff that made debugging more painful...